### PR TITLE
Add logic routing to appropriate UI when giving a rating

### PR DESCRIPTION
**Pull request by mchowning**
## Description
This PR checks whether a user is signed in, connected to the internet, and can rate a podcast (logic for being able to rate a podcast will probably be modified in the future), and based on that shows the appropriate placeholder screen.

In addition, this adds a Loading screen to display until we know whether the user can rate the podcast. This loading screen is the only UI in this PR that is not placeholder UI.

## Testing Instructions

1. Make sure that the ratings beta flag is turned on (it's on by default on debug builds)
3. Go to a podcast with a rating
4. Tap on the star rating
5. See that you are shown a screen for rating the podcast that mathes the designs (sQk9KlwngN1g8iJbQGAMva-fi-23_2052). Note that the stars portion of this screen is a placeholder that will be replaced in a follow-up PR.

## Checklist
- [X] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [X] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [X] I have considered whether it makes sense to add tests for my changes
- [X] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [X] Any jetpack compose components I added or changed are covered by compose previews

### CODE REVIEW

**Comment by mchowning** [comment]
Marking this as a draft so I can make some updates to the logic.

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/components/ratings/GiveRatingScreen.kt [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+        RowButton(
+            text = stringResource(LR.string.submit),
+            onClick = submitRating
+        )
+    }
+}
+
+@Composable
+private fun StarsRow(
```
Comment (mchowning) [inline modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/components/ratings/GiveRatingScreen.kt:111]: This `StarsRow` composable is just a placeholder. I'll be replacing it and the functionality around selecting stars in a follow-up PR.

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/components/ratings/GiveRatingScreen.kt [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+private fun Content(
+    state: GiveRatingViewModel.State.Loaded,
+    setStars: (Stars) -> Unit,
+    submitRating: () -> Unit,
+    onDismiss: () -> Unit,
+) {
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = Modifier.fillMaxSize()
+    ) {
```
Comment (ashiagr) [inline modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/components/ratings/GiveRatingScreen.kt:70]: Can we make it scrollable? I wasn't able to scroll the screen in landscape mode.

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/components/ratings/GiveRatingScreen.kt [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        ) {
+
+            PodcastCover(
+                uuid = state.podcastUuid,
+                coverWidth = 164.dp,
+            )
+
+            Spacer(Modifier.height(40.dp))
+
+            TextH30(text = stringResource(LR.string.podcast_rate, state.podcastTitle))
```
Comment (ashiagr) [inline modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/components/ratings/GiveRatingScreen.kt:91]: This text needs center-alignment and horizontal padding.

 
image

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/components/ratings/GiveRatingListenMoreScreen.kt [hunk 4]
```diff
@@ -4,6 +4,6 @@ import androidx.compose.runtime.Composable
 import au.com.shiftyjelly.pocketcasts.compose.components.TextH10
 
 @Composable
-fun GiveRatingListenMore() {
+fun GiveRatingListenMoreScreen() {
```
Comment (ashiagr) [inline modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/components/ratings/GiveRatingListenMoreScreen.kt:7]: This class looks redundant after this commit: https://github.com/Automattic/pocket-casts-android/pull/1451/commits/107de822cf4aeb0aef0f04d7847dcf5f5b03c4c2

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/viewmodel/GiveRatingViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    private val _state = MutableStateFlow<State>(State.Loading)
+    val state = _state.asStateFlow()
+
+    fun checkIfUserCanRatePodcast(
+        podcastUuid: String,
+        onFailure: (String) -> Unit
+    ) {
+        _state.value = State.Loading
+        viewModelScope.launch(Dispatchers.IO) {
+            val newState = getPodcastInfo(podcastUuid)
```
Comment (ashiagr) [inline modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/viewmodel/GiveRatingViewModel.kt:60]: This currently only depends on getting a valid podcast ID (and no listening history for the podcast as I see in [this commit](https://github.com/Automattic/pocket-casts-android/commit/d45673e57acf4e9acb94b8ff774f56fb9e806bac#diff-91e13144a4addcc841353c465d672b764a7fc5003723c965066c848d3377930dR72-R73)) and this condition is subject to change, correct?

We might want to add a small test for this condition whenever it is decided.

**Review by ashiagr — APPROVED** [review]
This looks great! ⭐ ⭐ ⭐ ⭐ ⭐ 
I just left minor nitpicks, feel free to merge it once those are addressed.

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/viewmodel/GiveRatingViewModel.kt (hunk 5 above)
Comment (mchowning) [reply modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/viewmodel/GiveRatingViewModel.kt:60]: Good suggestion. 👍  If I don't end up removing this logic (which is what I think may be what happens on my next commit), I'll look into adding a test.

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/components/ratings/GiveRatingListenMoreScreen.kt (hunk 4 above)
Comment (mchowning) [reply modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/components/ratings/GiveRatingListenMoreScreen.kt]: You're right, I forgot to remove that. 👍 Addressed in https://github.com/Automattic/pocket-casts-android/pull/1451/commits/0913b142c79454f31f14643183595d7e84ef85e8

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/components/ratings/GiveRatingScreen.kt (hunk 2 above)
Comment (mchowning) [reply modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/components/ratings/GiveRatingScreen.kt]: Thanks for catching that. Added in https://github.com/Automattic/pocket-casts-android/pull/1451/commits/ac82dc4015f7971d0bcb7f72b5d9d9d917458449

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/components/ratings/GiveRatingScreen.kt (hunk 3 above)
Comment (mchowning) [reply modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/components/ratings/GiveRatingScreen.kt]: Yes, yes it certainly does. Added in https://github.com/Automattic/pocket-casts-android/pull/1451/commits/de5bf3cb64df6e3536ff1b422d974e2365c8c73b

**Pull request merged**